base/array.jl

```diff
@@ -1115,35 +1115,75 @@ See [`sizehint!`](@ref) for notes about the performance model.
 See also [`vcat`](@ref) for vectors, [`union!`](@ref) for sets,
 and [`prepend!`](@ref) and [`pushfirst!`](@ref) for the opposite order.
 """
-function append!(a::Vector, items::AbstractVector)
+function append! end
+
+append!(a::Vector, items::AbstractVector) = try_append!(a, items)
+append!(a::Vector{T}, items::AbstractVector{U}) where {T, U <: T} = unsafe_append!(a, items)
+
+function unsafe_append!(a::Vector, items::AbstractVector)
     itemindices = eachindex(items)
     n = length(itemindices)
     _growend!(a, n)
     copyto!(a, length(a)-n+1, items, first(itemindices), n)
     return a
 end
+function try_append!(a::Vector, items::AbstractVector)
+    n = length(a)
+    try
+        unsafe_append!(a, items)
+    catch e
+        resize!(a, n)
+        rethrow(e)
```

Comment:
I think these should all be the 0-arg method:
```suggestion
        rethrow()
```
xref https://docs.julialang.org/en/v1/base/base/#Base.rethrow

Reply:
Good call! Done.